### PR TITLE
Release 0.4.0

### PR DESCRIPTION
**Pull request by dbrakenhoff**
Lots of improvements:
- Add `PasConnector` object for storing timeseries and models as Pastas .pas-files on disk. Simplest method to store pastas data on disk without any external dependencies (only `pastastore` and `pastas` required).
- Add method for detailed comparison of pastas Models: `pastastore.util.compare_models()`
- Add method to export all timeseries in a model as CSV-files.
- Add support for models using RechargeModel
- Add benchmarks
- Bug fixes:
  - Improve reading/writing zipfiles
  - Check for unsupported characters in metadata dictionaries
  - and more...

### CODE REVIEW

File: docs/examples.rst [hunk 1]
```diff
@@ -24,6 +24,22 @@ is stored in-memory (in dictionaries)::
...
+Using Pastas
+------------
+
+The following snippet shows how to use PastaStore with storage of 
+timeseries and models on disk as pas-files. This is the simplest implementation 
+that writes data to disk as no external dependencies are required::
+
+   import pastastore as pst
+
+   # define dict connector
```
Comment (OnnoEbbens) [inline docs/examples.rst:36]: Is the `PasConnector` replacing the `DictConnector`? The comment here refers to the `DictConnector` and the code to the `PasConnector`

File: docs/utils.rst [hunk 2]
```diff
@@ -2,9 +2,20 @@
...
-and pystore implementations.
+The `pastastore.util` submodule contains useful functions, i.e. for deleting
+databases, connector objects, and PastaStore objects or emptying a library of
+all its contents:
 
-* :meth:`pastastore.util.delete_arctic`
-* :meth:`pastastore.util.delete_pystore`
+
+* :meth:`pastastore.util.delete_pastastore`
+* :meth:`pastastore.util.delete_dict_connector`
```
Comment (OnnoEbbens) [inline docs/utils.rst:11]: what happens when you delete a connector? Why would you want to do that?

File: setup.py [hunk 3]
```diff
@@ -27,8 +27,7 @@
         'Programming Language :: Python :: 3.8'
     ],
     platforms='Windows, MacOS, *nix',
-    install_requires=['numpy>=1.15', 'pandas>=0.24', "tqdm",
-                      "pastas>=0.13", "arctic"],
+    install_requires=['numpy>=1.15', 'pandas>=0.24', "tqdm", "pastas>=0.13"],
```
Comment (OnnoEbbens) [inline setup.py:30]: I think this should have the same version requirements for pandas as in requirements.txt

Comment (OnnoEbbens) [reply setup.py:30]: Also, pastas should be a newer version right? There were some changes with the reading and writing of models

**Review by OnnoEbbens — APPROVED** [review]
Good job! Nice to see so many improvements. I left some comments in the code.

Is there a test for the new `pastastore.util.compare_models()` function? Maybe you can add this to `test_to_from_zip(prj)` to check if the models that you load from the zip are the same as the one you write to the zip.

File: docs/utils.rst (hunk 2 above)
Comment (dbrakenhoff) [reply docs/utils.rst:11]: It actually deletes the contents that the connector links to. Either a single library (oseries, stresses or models) or everything all at once. Perhaps a better name would clear that up, but don't have a better one yet.

**Pull request merged**